angular-ex-6/src/app/app.component.html

```diff
@@ -0,0 +1,93 @@
+<h1>Form Builder, Form Group, Form Control, Validators and ngsubmit</h1>
+
+<form class="form" [formGroup]="profileForm">
+  <div class="field">
+    <label class="label">username</label>
+    <input
+      id="name"
+      type="text"
+      class="input"
+      formControlName="name"
+      (click)="submit = false"
+      [class.invalid]="
+        profileForm.get('name')?.invalid &&
+        (profileForm.get('name')?.dirty || profileForm.get('name')?.touched)
+      "
+    />
+    <div
+      *ngIf="
+        profileForm.get('name')?.invalid &&
+        (profileForm.get('name')?.dirty || profileForm.get('name')?.touched)
+      "
+      class="error"
+    >
+      enter username
+    </div>
+  </div>
+  <div class="field">
+    <label class="label">email</label>
+    <input
+      type="text"
+      class="input"
+      formControlName="email"
+      (click)="submit = false"
+      [class.invalid]="
+        profileForm.get('email')?.invalid &&
+        (profileForm.get('email')?.dirty || profileForm.get('email')?.touched)
+      "
+    />
+    <div
+      *ngIf="
+        profileForm.get('email')?.invalid &&
```

Comment:
Create a function that checks all these conditions and returns only a boolean to simplify reading the html code